### PR TITLE
Add fix for find_transpose_contiguous_reshaper_unary in simplify_reshapes pass

### PR DESCRIPTION
**Pull request by nives-vukovic**
Expand unary make_op function in find_transpose_contiguous_reshaper_unary to include original unary operator attributes.

Resolves https://github.com/migraphx-benchmark/AMDMIGraphX/issues/166 https://github.com/ROCm/AMDMIGraphX/issues/2706

### CODE REVIEW

File: src/simplify_reshapes.cpp [hunk 1]
```diff
@@ -678,9 +678,12 @@ struct find_transpose_contiguous_reshaper_unary
         auto trans_ins     = r.instructions["trans_ins"];
         auto cont_ins      = r.instructions["cont_ins"];
         auto unary_op_name = ins->get_operator().name();
-        auto unary_ins     = m.insert_instruction(cont_ins, make_op(unary_op_name), trans_ins);
```
Comment (umangyadav) [inline src/simplify_reshapes.cpp:681]: This should be like 
`make_op(unary_op_name, ins->get_operator().to_value())`

Comment (nives-vukovic) [reply src/simplify_reshapes.cpp:681]: @umangyadav  Yes, I will fix it to this more general solution and update the PR description.

Comment (umangyadav) [reply src/simplify_reshapes.cpp:681]: can you add a unit- test ?

Comment (nives-vukovic) [reply src/simplify_reshapes.cpp:681]: @umangyadav I have added a unit test that represents the scenario that caused the issue in the model. It fails without the fix.

**Review by umangyadav — APPROVED** [review]

File: src/simplify_reshapes.cpp [hunk 2]
```diff
@@ -678,7 +678,8 @@ struct find_transpose_contiguous_reshaper_unary
...
+        auto unary_ins     = m.insert_instruction(
+            cont_ins, make_op(unary_op_name, ins->get_operator().to_value()), trans_ins);
```
Comment (pfultz2) [inline src/simplify_reshapes.cpp:682]: Dont use `make_op`. This can be written as `m.insert_instruction(cont_ins, ins->get_operator(), trans_ins)`.

Comment (nives-vukovic) [reply src/simplify_reshapes.cpp]: @pfultz2 I made the requested change in the code.

**Review by TedThemistokleous — APPROVED** [review]

**Pull request merged**